### PR TITLE
#10860: Refactored Conv2dConfig

### PR DESCRIPTION
**Pull request by sankarmanoj-tt**
### Ticket
#10860 

### Problem description
Refactored Conv2dConfig

### What's changed
Describe the approach used to solve the problem.
Summarize the changes made and its impact.

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp [hunk 1]
```diff
@@ -49,18 +45,13 @@ struct Conv2dConfig {
     bool enable_split_reader = false;
     bool enable_subblock_padding = false;
     static constexpr auto attribute_names = std::make_tuple(
```
Comment (mywoodstock) [inline ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp:47]: i think the `attribute_names` is no longer needed, we can delete.

Comment (sankarmanoj-tt) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp:47]: What about `attribute_values`?

Comment (mywoodstock) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp]: that as well

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_pybind.cpp [hunk 2]
```diff
@@ -57,9 +58,10 @@ void py_bind_conv2d(py::module& module) {
                 uint32_t groups,
                 std::optional<const ttnn::Tensor> bias_tensor,
                 std::optional<const Conv2dConfig> conv_config,
+                std::optional<const WormholeComputeKernelConfig> compute_config,
```
Comment (pavlejosipovic) [inline ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_pybind.cpp:61]: you can't bind WormholeComputeKernelConfig here, you need to use a variant check how matmul is doing that
```suggestion
                std::optional<const WormholeComputeKernelConfig> compute_config,
```

File: ttnn/cpp/ttnn/operations/core/compute_kernel/compute_kernel_config.cpp [hunk 3]
```diff
@@ -35,18 +35,27 @@ DeviceComputeKernelConfig init_device_compute_kernel_config(
...
+                        bool fp32_dest_acc_en = compute_kernel_config.fp32_dest_acc_en;
+                        bool packer_l1_acc = compute_kernel_config.packer_l1_acc;
+                        bool dst_full_sync_en = compute_kernel_config.dst_full_sync_en;
+                        defaultConfig = WormholeComputeKernelConfig{
+                            .math_fidelity = math_fidelity,
+                            .math_approx_mode = math_approx_mode,
+                            .fp32_dest_acc_en = fp32_dest_acc_en,
+                            .packer_l1_acc = packer_l1_acc,
+                            .dst_full_sync_en = dst_full_sync_en};
+                    } else if(arch == tt::ARCH::GRAYSKULL) {
```
Comment (pavlejosipovic) [inline ttnn/cpp/ttnn/operations/core/compute_kernel/compute_kernel_config.cpp:50]: pls remove this block grayskull is handled at line 28
```suggestion
                    } else if(arch == tt::ARCH::GRAYSKULL) {
```

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp [hunk 4]
```diff
@@ -26,19 +27,18 @@ namespace operations::conv {
...
     bool deallocate_activation = false;
     bool reallocate_halo_output = false;
-    uint32_t act_block_h_override = 0; // This argument is ignored when shard_layout == WIDTH_SHARDED.
-    uint32_t act_block_w_div = 1; //Amount by which the maximum possible act_block_width is divided. Max act_block_w = (in_channels * window_w * window_h)/total_num_cores;
-                                  //Ignored when shard_layout == HEIGHT_SHARDED or BLOCK_SHARDED
+
+    //Amount by which the maximum possible act_block is divided along a certain dimension.
+    //Divided along height for Height & Block Sharding. Divided along width for Width Sharding.
+    //When not set, chooses the smallest possible act block size(lowest perf). Setting to 1 chooses largest possible (highest perf)
+    std::optional<uint32_t> act_block_div = std::nullopt;
```
Comment (pavlejosipovic) [inline ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp:40]: it's odd that having 
- nullopt means optimize memory
- 1 means optimize perf
- 2 means actually use 2
Seem like we want another filed here which is true/false (optimize mem vs perf), and other optional filed which like 
```suggestion
    std::optional<uint32_t> act_block_div = std::nullopt;
```
ignore the first option I want exact value.

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_pybind.cpp (hunk 2 above)
Comment (sankarmanoj-tt) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_pybind.cpp:61]: I see no advantage in having different types for WormhomeComputeKernelConfig and GrayskullComputeKernelConfig. 

GrayskullComputeKernelConfig is a subset of WormhomeComputeKernelConfig. The utility functions that are used to read individual params from the ComputeConfig already take in a device arch and handle unused params appropriately. 

Instead of raising an error when there is a mismatch between device arch and compute config type, it makes more sense to just return the appropriate default.

The current approach forces the model writer to check the device arch and pass the appropriate type of ComputeConfig. This would be better handled inside the C++ op implementation.

Comment (pavlejosipovic) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_pybind.cpp:61]: All other ops work in this way, it's not ok for conv2d to do something else.
It's weird for users to have fixed wormhole confiig when they are for example working on Blackhole or Grayskull or some feature arch.

Comment (sankarmanoj-tt) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_pybind.cpp:61]: If this is the correct approach, it would be best to change conv first, and change the other ops later.

Comment (pavlejosipovic) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_pybind.cpp:61]: Pls keep conv inline with everything else.
If you do believe this is a better way to go, start a conversation about it on slack #tt-metal-developers or some other place.
If you reach an agreement that we should change this, make a plan and change it everywhere.

Comment (sankarmanoj-tt) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_pybind.cpp]: I've reverted these changes. As discussed, we'll stick with DeviceComputeKernelConfig.

File: ttnn/ttnn/operations/conv1d.py [hunk 5]
```diff
@@ -28,6 +28,7 @@ def Conv1d(
     groups: int = 1,
     bias_tensor: ttnn.Tensor = None,
     conv_config: Conv1dConfig = None,  # config overrides by user
+    compute_config: ttnn.WormholeComputeKernelConfig = None,
```
Comment (pavlejosipovic) [inline ttnn/ttnn/operations/conv1d.py:31]: should we use DeviceComputeKernelConfig

File: ttnn/ttnn/device.py [hunk 6]
```diff
@@ -145,4 +145,26 @@ def is_grayskull(device):
 pad_to_tile_shape = ttnn._ttnn.device.pad_to_tile_shape
 
 
+def GetComputeKernelConfig(
```
Comment (pavlejosipovic) [inline ttnn/ttnn/device.py:148]: add BH support

File: models/experimental/functional_unet/tt/unet_shallow_ttnn.py [hunk 7]
```diff
@@ -164,23 +169,23 @@ def __init__(
...
                 conv.use_activation_double_buffer if "use_activation_double_buffer" in conv else False
             ),
-            enable_split_reader=conv.use_split_reader if "use_split_reader" in conv else False,
             enable_subblock_padding=False,
             activation=activation,
             output_layout=ttnn.TILE_LAYOUT,
             input_channels_alignment=conv.input_channels_alignment if "input_channels_alignment" in conv else 32,
+            act_block_div=act_block_div,
+        )
+        self.conv_compute_config = ttnn.WormholeComputeKernelConfig(
```
Comment (pavlejosipovic) [inline models/experimental/functional_unet/tt/unet_shallow_ttnn.py:183]: why didn't you use GetComputeKernelConfig here?

File: models/experimental/functional_unet/tt/model_preprocessing.py [hunk 8]
```diff
@@ -63,12 +63,12 @@ def create_unet_model_parameters(model: unet_shallow_torch.UNet, input_tensor: t
         "num_cores_nhw": 55,
     }
 
-    parameters.c1["conv_blocking_and_parallelization_config_override"] = {"act_block_h": 16 * 32}
+    parameters.c1["conv_blocking_and_parallelization_config_override"] = {"act_block_div": 15}
```
Comment (pavlejosipovic) [inline models/experimental/functional_unet/tt/model_preprocessing.py:66]: why were values here changed? 16->15

File: models/demos/yolov4/ttnn/common.py [hunk 9]
```diff
@@ -74,21 +74,20 @@ def __call__(self, device, input_tensor):
...
-            packer_l1_accum_enabled=False,
             input_channels_alignment=16 if self.input_params[3] < 16 else 32,
             transpose_shards=False,
             reshard_if_not_optimal=self.reshard,
             deallocate_activation=self.deallocate,
             reallocate_halo_output=False,
         )
-        if self.act_block_h is not None:
-            conv_config.act_block_h_override = self.act_block_h
+        compute_config = ttnn.GetComputeKernelConfig(
```
Comment (pavlejosipovic) [inline models/demos/yolov4/ttnn/common.py:85]: why did you change all config values?

File: models/demos/wormhole/mamba/tt/mamba_conv.py [hunk 10]
```diff
@@ -54,11 +54,13 @@ def prepare_conv_config(self):
         self.conv1d_config = ttnn.Conv1dConfig(
             dtype=self.config.output_dtype,
             weights_dtype=self.config.weights_dtype,
-            math_fidelity=self.config.math_fidelity,
             shard_layout=ttnn.TensorMemoryLayout.HEIGHT_SHARDED,
             input_channels_alignment=32,
             deallocate_activation=True,
         )
+        self.conv1d_compute_config = ttnn.WormholeComputeKernelConfig(
```
Comment (pavlejosipovic) [inline models/demos/wormhole/mamba/tt/mamba_conv.py:61]: why don't you use the generic getter

File: models/demos/ttnn_resnet/tt/ttnn_functional_resnet50_new_conv_api.py [hunk 11]
```diff
@@ -323,22 +328,24 @@ def __call__(
...
                 else ttnn.TensorMemoryLayout.BLOCK_SHARDED,
                 reshard_if_not_optimal=reshard_if_not_optimal,
                 transpose_shards=transpose_shards,
-                packer_l1_accum_enabled=packer_l1_acc,
                 enable_act_double_buffer=enable_act_double_buffer,
                 enable_weights_double_buffer=True,
                 enable_split_reader=enable_split_reader,
                 enable_subblock_padding=enable_subblock_padding,
             ),
+            compute_config=ttnn.WormholeComputeKernelConfig(
```
Comment (pavlejosipovic) [inline models/demos/ttnn_resnet/tt/ttnn_functional_resnet50_new_conv_api.py:345]: GetComputeKernelConfig

**Review by pavlejosipovic — CHANGES_REQUESTED** [review]

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp [hunk 12]
```diff
@@ -30,19 +31,18 @@ using OutputWidth = uint32_t;
...
     bool deallocate_activation = false;
     bool reallocate_halo_output = false;
-    uint32_t act_block_h_override = 0; // This argument is ignored when shard_layout == WIDTH_SHARDED.
-    uint32_t act_block_w_div = 1; //Amount by which the maximum possible act_block_width is divided. Max act_block_w = (in_channels * window_w * window_h)/total_num_cores;
-                                  //Ignored when shard_layout == HEIGHT_SHARDED or BLOCK_SHARDED
+
+    //Amount by which the maximum possible act_block is divided along a certain dimension.
+    //Divided along height for Height & Block Sharding. Divided along width for Width Sharding.
+    //When not set, chooses the smallest possible act block size(lowest perf). Setting to 1 chooses largest possible (highest perf)
+    std::optional<uint32_t> act_block_div = 1;
```
[the remark this answers is not in the document]
Comment (sankarmanoj-tt) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp:44]: 1 still means use 1 as a value. It's just that 1 uses the largest possible act block size. null opt is the exception. Another option would be use -1 instead of nulllopt for using the smallest possible act block size.

[the remark this answers is not in the document]
Comment (pavlejosipovic) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp:44]: isn't for height sharding 1 the smallest possible value?

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp [hunk 13]
```diff
@@ -398,10 +409,8 @@ std::tuple<ttnn::Shape, ttnn::MemoryConfig, bool, bool> get_conv_padded_input_sh
         (!input_tensor_on_device || input_tensor_.is_sharded()) || conv_config.shard_layout.has_value(),
         "Tesor must be sharded or shard_layout must be set.");
 
-    TensorMemoryLayout shard_layout;
-    if (conv_config.shard_layout.has_value()) {
-        shard_layout = conv_config.shard_layout.value();
-    }
+    TT_FATAL(conv_config.shard_layout.has_value(),"Expects conv_config.shard_layout to be set.");
```
[the remark this answers is not in the document]
Comment (sankarmanoj-tt) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp:412]: shard_layout is optional and need not be set. `conv_config.shard_layout` has the actual value. I decided to use `conv_config.shard_layout` instead

File: models/experimental/functional_unet/tt/unet_shallow_ttnn.py (hunk 7 above)
Comment (sankarmanoj-tt) [reply models/experimental/functional_unet/tt/unet_shallow_ttnn.py:183]: Fixed.

File: models/demos/wormhole/mamba/tt/mamba_conv.py (hunk 10 above)
Comment (sankarmanoj-tt) [reply models/demos/wormhole/mamba/tt/mamba_conv.py:61]: Fixed.

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp [hunk 14]
```diff
@@ -53,54 +53,6 @@ struct Conv2dConfig {
     bool enable_weights_double_buffer = false; // Used on for block sharded convolutions
     bool enable_split_reader = false;
     bool enable_subblock_padding = false;
-    static constexpr auto attribute_names = std::make_tuple(
```
[the remark this answers is not in the document]
Comment (sankarmanoj-tt) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp:56]: @mywoodstock [said](https://github.com/tenstorrent/tt-metal/pull/14210#discussion_r1815797399) it's no long needed.

File: models/experimental/functional_unet/tt/model_preprocessing.py (hunk 8 above)
Comment (sankarmanoj-tt) [reply models/experimental/functional_unet/tt/model_preprocessing.py:66]: It was 16 tiles before, i.e. 512. Before act_block_h would specify the size of the exact size of the act_block. `act_block_div` specifies how many times the largest possible act_block must be divided. 

`largest_act_block_h = act_block_h * act_block_div`

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp (hunk 14 above)
[the remark this answers is not in the document]
Comment (pavlejosipovic) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp:56]: I'm using it, if you run tracy profile this will give you info on which setup conv was run with.
@mywoodstock is there maybe a better workflow that I don't know about?

File: models/demos/yolov4/ttnn/common.py (hunk 9 above)
Comment (sankarmanoj-tt) [reply models/demos/yolov4/ttnn/common.py:85]: I had copy-pasted the code snippet. I've fixed the config to be the same as the original.

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp (hunk 14 above)
[the remark this answers is not in the document]
Comment (sankarmanoj-tt) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp:56]: Are these attributes necessary for tracy to run correctly?

File: models/experimental/functional_unet/tt/model_preprocessing.py (hunk 8 above)
Comment (sankarmanoj-tt) [reply models/experimental/functional_unet/tt/model_preprocessing.py:66]: IMO, this approach makes it easier to choose a more performant value.

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp (hunk 14 above)
[the remark this answers is not in the document]
Comment (pavlejosipovic) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp:56]: No it will run w/o them, but when you look at the trace results you won't have info on these params that were used to run the conv, and I find that very helpful.
You have same thing on MMs

File: models/experimental/functional_unet/tt/model_preprocessing.py (hunk 8 above)
Comment (pavlejosipovic) [reply models/experimental/functional_unet/tt/model_preprocessing.py:66]: Maybe I'm missing something but how can a model writer assume the height of the block?

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp (hunk 13 above)
[the remark this answers is not in the document]
Comment (pavlejosipovic) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp:412]: shard_layout within the conv_config is optional
 std::optional<TensorMemoryLayout> shard_layout;
so you can't assume it has value? Am I missing something here?

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp [hunk 15]
```diff
@@ -790,7 +799,7 @@ static void adjust_conv_op_config_for_auto_shard(
         shard_orientation,
         compute_grid_size);
 
-    if (conv_config.act_block_h_override == 0 && conv_config.shard_layout != TensorMemoryLayout::WIDTH_SHARDED) {
+    if (conv_config.act_block_div == 1 && conv_config.shard_layout != TensorMemoryLayout::WIDTH_SHARDED) {
```
[the remark this answers is not in the document]
Comment (pavlejosipovic) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp:802]: bump

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp [hunk 16]
```diff
@@ -30,19 +31,18 @@ using OutputWidth = uint32_t;
...
     string activation = "";
     uint32_t input_channels_alignment = 32;
     bool deallocate_activation = false;
     bool reallocate_halo_output = false;
-    uint32_t act_block_h_override = 0; // This argument is ignored when shard_layout == WIDTH_SHARDED.
-    uint32_t act_block_w_div = 1; //Amount by which the maximum possible act_block_width is divided. Max act_block_w = (in_channels * window_w * window_h)/total_num_cores;
-                                  //Ignored when shard_layout == HEIGHT_SHARDED or BLOCK_SHARDED
+
+    //Amount by which the maximum possible act_block is divided along a certain dimension.
+    //Divided along height for Height & Block Sharding. Divided along width for Width Sharding.
```
[the remark this answers is not in the document]
Comment (pavlejosipovic) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp:42]: @mywoodstock bump?

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp (hunk 12 above)
[the remark this answers is not in the document]
Comment (pavlejosipovic) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp:44]: ok I see thx.

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp [hunk 17]
```diff
@@ -848,6 +858,7 @@ Result conv2d(
             input_tensor.layout());
     }
 
+    DeviceComputeKernelConfig compute_config = compute_config_.value_or(DeviceComputeKernelConfig());
```
[the remark this answers is not in the document]
Comment (pavlejosipovic) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp:861]: bump

File: ttnn/cpp/ttnn/operations/conv/conv2d/device/conv2d_op_sharded_program_factory.cpp [hunk 18]
```diff
@@ -2,7 +2,9 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+#include "device/device.hpp"
```
[the remark this answers is not in the document]
Comment (pavlejosipovic) [reply ttnn/cpp/ttnn/operations/conv/conv2d/device/conv2d_op_sharded_program_factory.cpp:5]: bump

File: ttnn/cpp/ttnn/operations/conv/conv2d/device/conv2d_op_sharded_program_factory.cpp [hunk 19]
```diff
@@ -1681,12 +1683,10 @@ operation::ProgramWithCallbacks multi_core_optimized_conv_sharded_v2_new(
     sliding_window::SlidingWindowConfig sliding_window_config,
     uint32_t output_channels,
     uint32_t groups,
-    bool untilize_out,
-    bool fuse_relu,
-    MathFidelity math_fidelity,
+    bool untilize_out, bool fuse_relu, MathFidelity math_fidelity,
```
[the remark this answers is not in the document]
Comment (pavlejosipovic) [reply ttnn/cpp/ttnn/operations/conv/conv2d/device/conv2d_op_sharded_program_factory.cpp:1686]: bump

File: ttnn/ttnn/device.py (hunk 6 above)
Comment (pavlejosipovic) [reply ttnn/ttnn/device.py:148]: bump

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp (hunk 14 above)
[the remark this answers is not in the document]
Comment (pavlejosipovic) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp:56]: Can we pls get this back?

**Comment by pavlejosipovic** [comment]
There is still a lot of references to act_block_h_override around the codebase that should be updated

File: ttnn/cpp/ttnn/operations/conv/conv2d/device/conv2d_op.hpp [hunk 20]
```diff
@@ -165,7 +165,7 @@ Tensor optimized_conv_new(const Tensor& a, const Tensor &b, std::optional<const
     DataType dtype,
     std::array<std::uint32_t, 4> input_tensor_shape,
     bool use_shallow_conv_variant,
-    std::optional<const DeviceComputeKernelConfig> compute_kernel_config = std::nullopt,
+    const DeviceComputeKernelConfig compute_kernel_config,
```
Comment (pavlejosipovic) [inline ttnn/cpp/ttnn/operations/conv/conv2d/device/conv2d_op.hpp:168]: pass by ref?

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp [hunk 21]
```diff
@@ -30,19 +31,18 @@ using OutputWidth = uint32_t;
...
+    //When not set, chooses the smallest possible act block size(lowest perf). Setting to 1 chooses largest possible (highest perf)
+    std::optional<uint32_t> act_block_div = 1;
+
```
Comment (pavlejosipovic) [inline ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp:45]: it doesn't make sense to have a filed std::optional and set default to a concret value,
do you agree?

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp [hunk 22]
```diff
@@ -790,7 +799,7 @@ static void adjust_conv_op_config_for_auto_shard(
         shard_orientation,
         compute_grid_size);
 
-    if (conv_config.act_block_h_override == 0 && conv_config.shard_layout != TensorMemoryLayout::WIDTH_SHARDED) {
+    if (conv_config.act_block_div == 1 && conv_config.shard_layout != TensorMemoryLayout::WIDTH_SHARDED) {
         if (in_channels <= constants::TILE_WIDTH / 2 && conv_config.input_channels_alignment == constants::TILE_WIDTH &&
             !is_mm_conv && conv_config.shard_layout == TensorMemoryLayout::HEIGHT_SHARDED && input_tensor_layout == Layout::ROW_MAJOR) {
```
Comment (pavlejosipovic) [inline ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp:804]: check on line 802 for width sharding should now be removed as act_block_div now affects all shardings.
Also this should check if act_block_div is nullptr (which should be the default), and than tweak it only if user didn't set it (if it was nullptr)

File: models/experimental/functional_unet/tt/unet_shallow_ttnn.py [hunk 23]
```diff
@@ -164,23 +169,23 @@ def __init__(
...
             weights_dtype=weights_dtype,
-            math_fidelity=ttnn.MathFidelity.LoFi,
             shard_layout=shard_layout,
             deallocate_activation=self.deallocate_activation,
-            fp32_dest_acc_enabled=True,
-            packer_l1_accum_enabled=False,
             enable_act_double_buffer=(
                 conv.use_activation_double_buffer if "use_activation_double_buffer" in conv else False
             ),
-            enable_split_reader=conv.use_split_reader if "use_split_reader" in conv else False,
```
Comment (pavlejosipovic) [inline models/experimental/functional_unet/tt/unet_shallow_ttnn.py:175]: why was enable_split_reader removed here?

File: models/demos/ttnn_resnet/tt/ttnn_functional_resnet50_new_conv_api.py [hunk 24]
```diff
@@ -569,25 +573,21 @@ def __init__(
...
             transpose_shards=self.transpose_shards,
-            packer_l1_accum_enabled=True if whb0_and_b16 else False,
+            act_block_div=2,
             enable_act_double_buffer=True if whb0_and_b16 else False,
             enable_split_reader=True if whb0_and_b16 or not is_wormhole_b0() else False,
             enable_subblock_padding=False,
             shard_layout=ttnn.TensorMemoryLayout.HEIGHT_SHARDED,
             reshard_if_not_optimal=False,
         )
-        if whb0_and_b16:
```
Comment (pavlejosipovic) [inline models/demos/ttnn_resnet/tt/ttnn_functional_resnet50_new_conv_api.py:585]: we really need to maintain this diff between galaxy and other systems to avoid di/dt issues, pls set equivalent setup with new args

File: models/demos/ttnn_resnet/tt/ttnn_functional_resnet50_new_conv_api.py [hunk 25]
```diff
@@ -730,11 +730,13 @@ def run(self, input_tensor, device, ops_parallel_config, conv_op_cache={}) -> tt
             input_height=self.conv1_input_height,
             input_width=self.conv1_input_width,
             conv_config=self.conv1_config,
+            compute_config=self.conv1_compute_config,
             conv_op_cache=conv_op_cache,
         )
         # Relu is fused with conv1
         if self.batch_size == 20:
             x = ttnn.reallocate(x)
+            act_block_div = (2,)
```
Comment (pavlejosipovic) [inline models/demos/ttnn_resnet/tt/ttnn_functional_resnet50_new_conv_api.py:739]: dead code?

**Review by tt-rkim — APPROVED** [review]
